### PR TITLE
(feat) Improve the visit header to match the design

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements
- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] My work includes tests or is validated by existing tests.

## Summary
This PR enhances the visit header by implementing UI improvements to align with the design specifications. Additionally, it abstracts the responsibility of rendering the visit-queue-details and patient-info to their respective components, thereby improving code maintenance.

Please note that the current implementation is not optimized for tablet screens. I will address the tablet view optimization in another PR and follow up accordingly.
## Screenshots
Kapture 2023-05-22 at 12 07 14

### CODE REVIEW

File: packages/esm-patient-chart-app/src/visit-header/visit-queue-details.component.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      <span className={styles.patientInfo}>
+        {getServiceString()}
+        <Tag
+          className={priority === 'Priority' ? styles.priorityTag : styles.tag}
+          type={getTagType(priority?.toLowerCase())}
+        >
+          {priority}
+        </Tag>
+        <Button
+          iconDescription={t('moveToNextService', 'Move to next service')}
```
Comment (CynthiaKamau) [inline packages/esm-patient-chart-app/src/visit-header/visit-queue-details.component.tsx:52]: Any reason why you removed the text?

**Review by CynthiaKamau — DISMISSED** [review]

File: packages/esm-patient-chart-app/src/visit-header/visit-queue-details.component.tsx (hunk 1 above)
Comment (donaldkibet) [reply packages/esm-patient-chart-app/src/visit-header/visit-queue-details.component.tsx:52]: Making match the design

File: packages/esm-patient-chart-app/src/visit-header/visit-queue-details.test.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+describe('VisitQueueDetail', () => {
+  const mockQueueEntry = {
+    service: 'Test Service',
+    status: STATUS.IN_SERVICE,
+  };
+
+  const mockPriority = PRIORITY.EMERGENCY;
+
+  it('renders without crashing', () => {
+    const { container } = render(<VisitQueueDetail queueEntry={mockQueueEntry} priority={mockPriority} />);
```
Comment (denniskigen) [inline packages/esm-patient-chart-app/src/visit-header/visit-queue-details.test.tsx:20]: Nit: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-container-to-query-for-elements

File: packages/esm-patient-chart-app/src/visit-header/patient-info-card.test.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+describe('PatientInfoCard', () => {
+  const mockPatientMale = {
+    name: [{ given: ['John'], family: 'Doe' }],
+    birthDate: '1990-01-01',
+    gender: 'male',
+  };
+
+  const mockPatientFemale = {
+    ...mockPatientMale,
```
Comment (denniskigen) [inline packages/esm-patient-chart-app/src/visit-header/patient-info-card.test.tsx:19]: (Really tiny nit): It'd be nice if `John` got a female name here.

**Review by denniskigen — APPROVED** [review]

**Comment by denniskigen** [comment]
Sorry for not asking sooner, but shouldn't the visit header automatically pick up changes on mutate? I notice that in the video you had to reopen the patient chart to get the visit header changes to reflect. Is this a known issue?

**Comment by brandones** [comment]
@donaldkibet @denniskigen Can we either get this in or close it out?

Also what are the designs this is referring to?

**Comment by donaldkibet** [comment]
Link to the associated designs closing this now https://app.zeplin.io/project/60d59321e8100b0324762e05/screen/61bb8a40b826af607ca2daea

**Pull request closed without merging**